6th task - DI & Interfaces

alspetrova opened the pull request:
Application designed for testability and reuse

-  non-static Controller with Service DI
-  non-static Service with Repo DI
-  Service with Interface
-  non-static Repository with Interface
-  non-static Loan Request
-  non-static Loan Response with encapsulated autogenerated incremental id
-  Loan Response encapsulates Response Type enum: APPROVED, DENIED
-  Self-made tests implemented (with main class)

alspetrova committed:
6th task - DI & Interfaces

alspetrova committed:
6th task - DI & Interfaces fix

AndrewCharykov reviewed (changes requested):
1)Создай  в app папки controller, enums, logger, model, repository, service и перенеси классы в соответствующее место

2)Метод save в LoanCalcRepository должен возвращать объект LoanResponse

3)Дальше нужно переименовать  StaticVariableLoanCalcRepository и поправить реализацию метода save

4)Поправить реализацию метода createRequest в LoanCalcService

alspetrova committed:
6th task - DI & Interfaces fix (2)
1)Создала в app папки controller, enums, logger, model, repository, service и перенесла классы
2)Метод save в LoanCalcRepository возвращает объект LoanResponse
3)Переименовала StaticVariableLoanCalcRepository и поправила реализацию метода save
4)Поправила реализацию метода createRequest в LoanCalcService

alspetrova committed:
6th task - DI & Interfaces fix (2)
1)Создала в app папки controller, enums, logger, model, repository, service и перенесла классы
2)Метод save в LoanCalcRepository возвращает объект LoanResponse
3)Переименовала StaticVariableLoanCalcRepository и поправила реализацию метода save
4)Поправила реализацию метода createRequest в LoanCalcService

alspetrova committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/main/java/com/tinkoff/edu/app/controller/LoanCalcController.java
#	src/main/java/com/tinkoff/edu/app/enums/LoanType.java
#	src/main/java/com/tinkoff/edu/app/enums/ResponseType.java
#	src/main/java/com/tinkoff/edu/app/logger/LoanCalcLogger.java
#	src/main/java/com/tinkoff/edu/app/model/LoanRequest.java
#	src/main/java/com/tinkoff/edu/app/model/LoanResponse.java
#	src/main/java/com/tinkoff/edu/app/service/IpNotFriendlyService.java
#	src/main/java/com/tinkoff/edu/app/service/LoanCalcService.java
#	src/main/java/com/tinkoff/edu/app/service/LoanCalcServiceInterface.java

alspetrova committed:
fix - убрала создание объекта из LoanCalcController.java
создание объектов в main
